### PR TITLE
Fix memleak in db/esil/apple

### PR DESCRIPTION
**Pull request by eagleoflqj**
**Checklist**

- [ ] Closing issues: #issue
- [ ] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [radare2book](https://github.com/radareorg/radare2book)

**Description**

#18150 
As I said in discord, there's a problem of `r_skiplist_insert`. When it meets duplicate key, it simply returns the original node. but this result is always ignored, thus a call `r_skiplist_insert (list, data)` may cause memory leak.
Should we check the result every time, or use a new function to insert and free if duplicate?
the leak can be reproduced by `r2 -Qc 'pd 2~?push' bins/mach0/arm16-ofp.lzh`

### CODE REVIEW

**Comment by trufae** [comment]
Do you think we can merge this PR as is? or you want to address this skiplist issue you mention in te same pr?

**Comment by eagleoflqj** [comment]
> Do you think we can merge this PR as is? or you want to address this skiplist issue you mention in te same pr?

You mean add the api to automatically free? If you think it’s the right way, then i’ll do it in this pr.

File: libr/util/skiplist.c [hunk 1]
```diff
@@ -198,6 +198,15 @@ R_API RSkipListNode* r_skiplist_insert(RSkipList* list, void* data) {
 	return x;
 }
 
+R_API bool r_skiplist_insert_autofree(RSkipList* list, void* data) {
+	RSkipListNode* node = r_skiplist_insert (list, data);
+	if (node && data != node->data) { // duplicate
+		free (data);
```
Comment (trufae) [inline libr/util/skiplist.c:204]: uhm maybe this should be calling the list.free function pointer instead of just free

**Review by trufae — CHANGES_REQUESTED** [review]

File: libr/util/skiplist.c (hunk 1 above)
Comment (eagleoflqj) [reply libr/util/skiplist.c:204]: right

**Pull request merged**